cerberus/client.py

```diff
@@ -291,13 +317,14 @@ def update_sdb(self, sdb_id, owner=None, description=None, user_group_permission
 
     def delete_secret(self, vault_path):
         """Delete a secret from the given vault path"""
-        secret_resp = requests.delete(self.cerberus_url + '/v1/secret/' + vault_path,
+        secret_resp = requests.delete(self.cerberus_url + '/v1/secret/' + str.strip(vault_path, '/'),
```

Comment:
But they never actually have a '/' at the end (or beginning) of a path, which is what that strip is taking care of.  I put in the same logic elsewhere, where there were situations where someone could mistakenly leave a trailing '/', which would cause an error.

In the case of deletion, with a key in a virtual path, you cannot delete the virtual path directly.  If you delete all of the keys underneath a virtual path, then the virtual path is automatically deleted as well.

so 
```python
client.put_secret('category/sdb/virt/key', {'key': 'value in virtual path'})
client.list_secret('category/sdb/')
 # returns something like
 # {'......
 #   'data': {'keys': ['virt/']},
 # ......}
client.list_secret('category/sdb/virt/')
# Returns
 # {'......
 #   'data': {'keys': ['key']},
 # ......}

# this is the behavior before adding the stripping
client.delete_secret('category/sdb/virt/')
# ERROR

client.delete_secret('category/sdb/virt')
# ERROR

client.delete_secret('category/sdb/virt/key')
# Success

client.list_secret('category/sdb/')
 # returns something like
 # {'......
 #   'data': {'keys': []},
 # ......}
```

What the strip command helps prevent is when someone issues a
```python
client.delete_secret('category/sdb/virt/key/')
```
Which originally will cause an error, will just delete the key properly after the change.


Now with that all said..  I'm OK with removing the strip command from delete_secret, since someone is much more likely to be more specific when they are deleting a key.

Reply:
I went back and did some testing both pre and post change and it didn't seem to negatively affect the process of deletion.

Though I highly encourage others to test that.  I may be missing a corner case.

Reply:
Hmm I found an fascinating corner case.
With the code as it is.  You can create a blank secret.  Because I don't do any stripping on put_secret, you can do things like 
```python
client.put_secret('category/sdb/virt/', {'key': "The path to this secret is '' "})
```

So before I put in the stuff to do stripping, you could at still access the secret by leaving a trailing '/'.  Since I do the stripping for '/' in get_secrets, list_secrets, and delete_secret you can't actually access the blank key anymore.

While I'd love to prevent the creation of blank secrets, I feel like because the API and other libraries allow the creation of blank keys, I don't think I should be introducing a regression in features.